query-service-impl/src/main/java/org/hypertrace/core/query/service/projection/ProjectionTransformation.java

```diff
@@ -315,6 +322,82 @@ private Filter rebuildFilterOmittingDefaults(
     return builder.clearChildFilter().addAllChildFilter(childFilters).build();
   }
 
+  private QueryRequest rebuildRequest(
+      QueryRequest original,
+      List<Expression> selections,
+      List<Expression> aggregations,
+      Filter filter,
+      List<Expression> groupBys,
+      List<OrderByExpression> orderBys) {
+
+    Set<AttributeExpression> attributeExpressionSet =
+        Stream.concat(
+                findAttributeExpressionFromOrderBy(orderBys).stream(),
+                findAttributeExpressionFromFilter(filter).stream())
+            .collect(Collectors.toSet());
+
+    return rebuildRequestOmittingDefaults(
+        original,
+        selections,
```

Comment:
We had also discussed replacing columnIdentifiers with attributeExpressions in preprocessing. Up to you if you want to do it, but it may be easier to do that here than support both in the transformation.

Reply:
Can we do that in a follow up PR ?

Reply:
yes

Reply:
Added in #100

Reply:
resolved for now